hazelcast-wm/src/test/java/com/hazelcast/wm/test/WebFilterClientFailOverTests.java

```diff
@@ -0,0 +1,97 @@
+package com.hazelcast.wm.test;
+
+import com.hazelcast.config.FileSystemXmlConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.IMap;
+import org.apache.http.client.CookieStore;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.junit.Test;
+
+import java.io.File;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class WebFilterClientFailOverTests extends AbstractWebFilterTest {
+
+    String testName;
+    protected WebFilterClientFailOverTests(String serverXml1) {
+        super(serverXml1);
+    }
+
+    protected WebFilterClientFailOverTests(String serverXml1, String serverXml2) {
+        super(serverXml1, serverXml2);
+        testName = serverXml1;
+    }
+
+    @Test
+    public void whenClusterIsDown_enabledDeferredWrite() throws Exception {
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
```

Comment:
This test is doing too much. You should not do any assert in the initialization of the test. 

A bit cleaner is:
```
 @Test
    public void whenClusterIsDown_enabledDeferredWrite() throws Exception {
        CookieStore cookieStore = new BasicCookieStore();
        executeRequest("write", serverPort1, cookieStore);
        executeRequest("read", serverPort1, cookieStore);
        
	hz.shutdown();
        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
        assertEquals("true", executeRequest("remove", serverPort1, cookieStore));
        assertEquals("null", executeRequest("read", serverPort1, cookieStore));
     
        hz = Hazelcast.newHazelcastInstance(
                new FileSystemXmlConfig(new File(sourceDir + "/WEB-INF/", "hazelcast.xml")));
        Thread.sleep(8000);
        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
    }
```

And I'm still not that happy because of the assert immediately after the hz.shutdown. Are these the part of the logic to be tested or are you still setting up your system. Imho you are still setting up your system and therefor they should not be asserts. Please check the wiki:
https://hazelcast.atlassian.net/wiki/display/EN/Testing+Practices